Src/Campus.Infrastructure.Data/Repositories/UnitOfWork.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System.Data;
+using Campus.Domain.Interfaces.Interfaces;
+
+namespace Campus.Infrastructure.Data.Repositories
+{
+    public class UnitOfWork : IUnitOfWork
+    {
+        public UnitOfWork(IDbConnection connection)
+        {
+            connection.Open();
+            _connection = connection;
+        }
+
+        private IDbConnection _connection;
+        private IDbTransaction _transaction;
+
+        public IDbConnection Connection => _connection;
+
+        public IDbTransaction Transaction => _transaction;
+
+        public void Begin()
+        {
+            _transaction = _connection.BeginTransaction();
+        }
+
+        public void Commit()
+        {
+            _transaction.Commit();
+            Dispose();
```

Comment:
DI container performs object dispose after every scoped execution, you don't need to do it manually